enterprise/views.py

```diff
@@ -686,19 +803,26 @@ def get(self, request, enterprise_uuid, course_id):
             * No enterprise customer uuid kwarg `enterprise_uuid` in request.
             * No enterprise customer found against the enterprise customer
                 uuid `enterprise_uuid` in the request kwargs.
+            * No course is found in database against the provided `course_id`.
         """
         # Verify that all necessary resources are present
         verify_edx_resources()
 
         enterprise_customer, course, modes = self.get_base_details(enterprise_uuid, course_id)
 
+        # Create a link between the user and the enterprise customer if it
+        # does not already exist.
+        enterprise_customer_user, __ = EnterpriseCustomerUser.objects.get_or_create(
+            enterprise_customer=enterprise_customer,
+            user_id=request.user.id
+        )
+
         enrollment_client = EnrollmentApiClient()
         enrolled_course = enrollment_client.get_course_enrollment(request.user.username, course_id)
-
-        if (enrolled_course is not None or
+        if (enrolled_course is not None and
                 EnterpriseCourseEnrollment.objects.filter(
-                    enterprise_customer_user__enterprise_customer=enterprise_customer,
-                    enterprise_customer_user__user_id=request.user.id,
+                    enterprise_customer_user__enterprise_customer=enterprise_customer_user.enterprise_customer,
+                    enterprise_customer_user__user_id=enterprise_customer_user.user_id,
```

Comment:
Yeah, we don't seem to be doing anything with the above EnterpriseCustomerUser here other than substituting it here, so I don't see what the need is.